backend/Generator/main.py

```
@@ -35,102 +36,75 @@ def __init__(self):
-        if len(keyword_sentence_mapping.keys()) == 0:
-            return final_output
-        else:
-            generated_questions = generate_normal_questions(keyword_sentence_mapping, self.device, self.tokenizer, self.model)
+        accuracy = self.get_accuracy()
 
-        final_output["statement"] = modified_text
-        final_output["questions"] = generated_questions["questions"]
-        
-        if self.device.type == 'cuda':
-            torch.cuda.empty_cache()
+        self.current_difficulty = adjust_difficulty(
```

Ruff (0.15.7)

[error] Undefined name `adjust_difficulty`

(F821)